### PR TITLE
Screen peek/poke functions as in the "extending" doc

### PR DESCRIPTION
**Pull request by csboling**
Adds `s = screen.peek(x, y, w, h)` and `screen.poke(x, y, w, h, s)` functions. The `peek` function is the example used in https://github.com/monome/docs/pull/274 and there's been some interest in the discord study group in interacting with the frame buffer contents this way. Demo script [here](https://github.com/csboling/cobbles).

### CODE REVIEW

**Review by ngwese — COMMENTED** [review]
a general question w.r.t. use cases for peek/poke - if we exposed cairo image surfaces such that one could allocation a surface, draw into it, and then composite in at a given location (via `cairo_set_source_surface(...)` + `cairo_paint(...)`) would there be a need for the peek/poke operations?

File: matron/src/hardware/screen.c [hunk 1]
```diff
@@ -449,5 +450,52 @@ extern void screen_export_png(const char *s) {
...
+
+    (void)buf;
+
+    uint32_t *data = (uint32_t *)cairo_image_surface_get_data(surface);
+    if (!data) {
+        return;
+    }
+    uint8_t *p = buf;
+    uint32_t pixel;
+    for (int j = y; j < y + h; j++) {
```
[the remark this answers is not in the document]
Comment (csboling) [reply matron/src/hardware/screen.c:489]: There is bounds checking in the weaver functions https://github.com/monome/norns/pull/1246/files#diff-49d5275bded798431a6b47e12ebb6b0cb0ff4be76141234beab356ca786e71d5R882 but actually all existing `screen` coordinates are 0-indexed, e.g.

```lua
screen.level(15)
screen.pixel(0, 0)
screen.fill()
screen.update()
```

File: matron/src/hardware/screen.c [hunk 2]
```diff
@@ -449,5 +450,52 @@ extern void screen_export_png(const char *s) {
...
+    }
+    return buf;
+}
+
+void screen_poke(int x, int y, int w, int h, unsigned char *buf) {
+    CHECK_CR
+    w = (w <= (128 - x)) ? w : (128 - x);
+    h = (h <= (64 - y))  ? h : (64 - y);
+
+    (void)buf;
```
[the remark this answers is not in the document]
Comment (csboling) [reply matron/src/hardware/screen.c]: Oops, fixed in 7969501

**Comment by csboling** [comment]
> if we exposed cairo image surfaces

Yeah we could potentially expose a more generalized cairo interface and then implement this kind of thing Lua-side, which could 
be nice if there are a lot more use cases (other recent PRs have talked about sprite sheets and the like). Prior art includes https://luapower.com/cairo. Mainly I wanted to close the loop here on this feature that's used as an example in the https://monome.org/docs/norns/extending tutorial and this is comparatively simpler to implement. We could always re-implement the Lua `screen.peek` / `screen.poke` functions using a future generalized Cairo wrapper.

File: matron/src/hardware/screen.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ngwese) [reply matron/src/hardware/screen.c:487]: thank you for setting me straight, I had missed the bounds checking in the weaver functions.

**Review by ngwese — APPROVED** [review]
LGTM

**Pull request merged**